scaffoldmaker/utils/eftfactory_tricubichermite.py

```diff
@@ -173,6 +173,354 @@ def createEftShellApexTop(self, nodeScaleFactorOffset0, nodeScaleFactorOffset1):
+            remapEftNodeValueLabel(eft, [ ln ], Node.VALUE_LABEL_D_DS2, [ (Node.VALUE_LABEL_D_DS1, [so + 1, so + 5]), (Node.VALUE_LABEL_D_DS2, [so + 2, so + 5]), (Node.VALUE_LABEL_D_DS3, [1, so + 4]) ])
+            # 2 terms for cross derivative 1 2 to correct circular apex: -sin(theta).phi, cos(theta).phi
+            remapEftNodeValueLabel(eft, [ ln ], Node.VALUE_LABEL_D2_DS1DS2, [ (Node.VALUE_LABEL_D_DS1, [1, so + 2, so + 3 , so + 5]), (Node.VALUE_LABEL_D_DS2, [so + 1, so + 3, so + 5]) ])
+            # zero other cross derivative parameters
+            remapEftNodeValueLabel(eft, [ ln ], Node.VALUE_LABEL_D2_DS1DS3, [])
+            remapEftNodeValueLabel(eft, [ ln ], Node.VALUE_LABEL_D2_DS2DS3, [])
+            remapEftNodeValueLabel(eft, [ ln ], Node.VALUE_LABEL_D3_DS1DS2DS3, [])
+
+            ln = layer*4 + 2
+            # 3 terms for d/dxi2 via general linear map:
+            remapEftNodeValueLabel(eft, [ ln ], Node.VALUE_LABEL_D_DS2, [ (Node.VALUE_LABEL_D_DS1, [so + 6, so + 10] ), (Node.VALUE_LABEL_D_DS2, [so + 7, so + 10]), (Node.VALUE_LABEL_D_DS3, [1, so + 9]) ])
+            # 2 terms for cross derivative 1 2 to correct circular apex: -sin(theta).phi, cos(theta).phi
+            remapEftNodeValueLabel(eft, [ ln ], Node.VALUE_LABEL_D2_DS1DS2, [ (Node.VALUE_LABEL_D_DS1, [1, so + 7, so + 8, so + 10]), (Node.VALUE_LABEL_D_DS2, [so + 6, so + 8, so + 10]) ])
+            # zero other cross derivative parameters
+            remapEftNodeValueLabel(eft, [ ln ], Node.VALUE_LABEL_D2_DS1DS3, [])
+            remapEftNodeValueLabel(eft, [ ln ], Node.VALUE_LABEL_D2_DS2DS3, [])
+            remapEftNodeValueLabel(eft, [ ln ], Node.VALUE_LABEL_D3_DS1DS2DS3, [])
+
+        ln_map = [ 1, 1, 2, 3, 1, 1, 4, 5 ]
+        remapEftLocalNodes(eft, 5, ln_map)
+        assert eft.validate(), 'eftfactory_tricubichermite.createEftPyramidTop:  Failed to validate eft'
+        return eft
+
+    def createEftPyramidTop(self, nodeScaleFactorOffset0, nodeScaleFactorOffset1, nodeScaleFactorOffsetUp):
+        '''
+        Create a tricubic hermite element field for a solid pyramid for the axis of a
+        solid sphere pole, with xi1 and xi3 collapsed on xi2 = 1. Each collapsed node
+        has 5 scale factors giving the cos, sin coefficients of the radial line from
+        global derivatives, plus the arc subtended by the element in radians, so the
+        circumferential direction is rounded, cos and sin coefficients of the inclination
+        angle. Need to create a new template for each sector around axis giving common
+        nodeScaleFactorOffset values on common faces. Suggestion is to start at 0 and
+        add 100 for each radial line around axis.
+        :param nodeScaleFactorOffset0: offset of node scale factors at axis on xi1=0
+        :param nodeScaleFactorOffset1: offset of node scale factors at axis on xi1=1
+        :param nodeScaleFactorOffsetUp: offset of node scale factors as inclination increases
```

Comment:
offset of first scale factor for inclination at pole, increasing by 2 with each layer away from axis. Suggest starting at 100000 on axis of sphere.